packages/flutter/lib/src/material/navigation_bar.dart

```diff
@@ -148,14 +148,17 @@ class NavigationBar extends StatelessWidget {
       ?? navigationBarTheme.labelBehavior
       ?? defaults.labelBehavior!;
     final double additionalBottomPadding = MediaQuery.of(context).padding.bottom;
+    final double additionalLeftPadding = MediaQuery.of(context).padding.left;
+    final double additionalRightPadding = MediaQuery.of(context).padding.right;
 
     return Material(
       color: backgroundColor
         ?? navigationBarTheme.backgroundColor
         ?? defaults.backgroundColor!,
       elevation: elevation ?? navigationBarTheme.elevation ?? defaults.elevation!,
       child: Padding(
-        padding: EdgeInsets.only(bottom: additionalBottomPadding),
+        padding: EdgeInsets.fromLTRB(additionalLeftPadding, 0,
+          additionalRightPadding, additionalBottomPadding),
         child: MediaQuery.removePadding(
           context: context,
           removeBottom: true,
```

Comment:
Should we set `removeLeft` and `removeRight` here?

Reply:
Looks like that padding is getting consumed above, so it should be removed for ancestors as suggested.

Reply:
Thanks for the comments! Looks like SafeArea did same thing just like Padding() and MediaQuery.removePadding(), so I removed both and use SafeArea instead. Let me know if you have any suggestions.
https://github.com/flutter/flutter/blob/5464c5bac7/packages/flutter/lib/src/widgets/safe_area.dart#:~:text=return%20Padding(,child%3A%20MediaQuery.removePadding(